code/datalab-app/src/app/PrivateApp.js

```diff
@@ -10,14 +11,33 @@ import PublishingPage from './pages/PublishingPage';
 import DaskPage from './pages/DaskPage';
 import SparkPage from './pages/SparkPage';
 import ModalRoot from './containers/modal/ModalRoot';
+import RoutePermissions from './components/common/RoutePermissionWrapper';
 
-const PrivateApp = () => (
-  <NavigationContainer>
+const PrivateApp = ({ promisedUserPermissions }) => (
+  <NavigationContainer userPermissions={promisedUserPermissions.value}>
     <Switch>
       <Route exact path="/" component={LandingPage} />
-      <Route exact path="/storage" component={DataStoragePage} />
-      <Route exact path="/notebooks" component={NotebooksPage} />
-      <Route exact path="/publishing" component={PublishingPage} />
+      <RoutePermissions
+        exact
+        path="/storage"
+        component={DataStoragePage}
+        promisedUserPermissions={promisedUserPermissions}
+        permission="project:storage:list"
```

Comment:
Do we want the list of all permissions to be in a single const file and then imported? If you agree I suggest that we merge this PR first and then refactor.